src/main/java/org/mule/extension/aggregator/internal/storage/content/SimpleAggregatedContent.java

```diff
@@ -53,24 +61,86 @@ public void add(TypedValue newContent, Long timeStamp) {
 
   @Override
   public void add(TypedValue newContent, Long timeStamp, int sequenceNumber) {
-    sequencedElements.put(sequenceNumber, newContent);
+    SequencedElement sequencedElement;
+    if (sequencedElements.containsKey(sequenceNumber)) {
+      sequencedElement = (SequencedElement) sequencedElements.get(sequenceNumber).getValue();
+    } else {
+      sequencedElement = new SequencedElement();
+    }
+    sequencedElement.add(newContent);
+    sequencedElements.put(sequenceNumber, TypedValue.of(sequencedElement));
+
     updateTimes(timeStamp);
   }
 
   @Override
   public List<TypedValue> getAggregatedElements() {
     List<TypedValue> orderedElements = new ArrayList<>();
     if (sequencedElements.size() > 0) {
-      orderedElements = sequencedElements.entrySet().stream().sorted(comparingInt(Map.Entry::getKey)).map(Map.Entry::getValue)
+      orderedElements = sequencedElements.entrySet().stream().sorted(comparingInt(Map.Entry::getKey))
+          .flatMap(val -> ((SequencedElement) val.getValue().getValue()).get().stream())
           .collect(toList());
     }
     orderedElements.addAll(unsequencedElements);
     return orderedElements;
   }
 
-
   public boolean isComplete() {
-    return maxSize == sequencedElements.size() + unsequencedElements.size();
+    Integer sequencedElementsSize = sequencedElements.values().stream().map(v -> ((SequencedElement) v.getValue()).size())
```

Comment:
doesn't this cast affect serialization in any way if you have elements from previous versions of the connector? Shouldn't it verify if it was upgraded?

Reply:
It shouldn't because the upgradeIfNeeded() method is being called at the start() of the AbstractAggregatorExecutor. So the elements from the previous version of the connector are upgraded before this cast.